codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieDecoder.java

```diff
@@ -32,27 +32,49 @@
     private final InternalLogger logger = InternalLoggerFactory.getInstance(getClass());
 
     private final boolean strict;
+    private final boolean reportException;
+
+    protected CookieDecoder(boolean strict, boolean reportException) {
+       this.strict = strict;
+       this.reportException = reportException;
+    }
 
     protected CookieDecoder(boolean strict) {
-        this.strict = strict;
+        this(strict, false);
     }
 
     protected DefaultCookie initCookie(String header, int nameBegin, int nameEnd, int valueBegin, int valueEnd) {
         if (nameBegin == -1 || nameBegin == nameEnd) {
-            logger.debug("Skipping cookie with null name");
+            if (logger.isDebugEnabled()) {
```

Comment:
The check for `isDebugEnabled()` is redundant in the such cases. I think, it's should be logging or exception thrown (not both):
```java
private final boolean needReport;

protected CookieDecoder(boolean strict, boolean reportException) {
    this.strict = strict;
    this.reportException = reportException;
    needReport = reportException || logger.isDebugEnabled();
}

private final void reportError(String message) {
    if (reportException) {
        throw new IllegalArgumentException(message);
    } else {
        logger.debug(message);
    }
}

...
if (needReport) {
    reportError("Some error " + xxx + " ...");
}
```